jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -85,6 +91,21 @@ def __init__(self, *args, **kwargs):
         self.subset_states = []
         self.spectral_display_unit = None
 
+        # if hasattr(self.app, '_link_type'):
+        #     self.display_sky_coordinates = (self.app._link_type == 'wcs')
+        # else:
+        #     self.display_sky_coordinates = False
+
+    def _on_link_update(self, *args):
+        # set to True only when link type is 'wcs'. if none or 'pixel', set
+        # this to false and display pixel coordinates
+
+        self.display_sky_coordinates = (self.app._link_type == 'wcs')
+
+        # only update if link changes and there is an active selection in plugin
+        if self.subset_selected != self.subset_select.default_text:
+            self._get_subset_definition(*args)
```

Comment:
`self.subset_selected` can be a list in multiselect mode, so this may need some more checks.